langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/ToolExecutionHelper.java

```diff
@@ -20,7 +20,15 @@ static String extractResult(JsonNode result) {
         if (result.has("result")) {
             JsonNode resultNode = result.get("result");
             if (resultNode.has("content")) {
-                return extractSuccessfulResult((ArrayNode) resultNode.get("content"));
+                String content = extractSuccessfulResult((ArrayNode) resultNode.get("content"));
+                boolean isError = false;
+                if (resultNode.has("isError")) {
+                    isError = resultNode.get("isError").asBoolean();
+                }
+                if (isError) {
+                    content = String.format(EXECUTION_ERROR_MESSAGE + ". %s", content);
```

Comment:
I think it would be clearer if we used `:` instead of `.` between these two strings. Or maybe, even better, something like `EXECUTION_ERROR_MESSAGE + ". The tool returned: %s"` 

Maybe we should make this whole thing customizable (add an SPI for applications to plug in a Supplier of error messages) but that will be a separate enhancement I guess

Reply:
@jmartisk , thank-you. I updated as you suggested.